tests/integration/test_package_managers.py

```diff
@@ -161,3 +123,66 @@ def test_packages(
         )
         log.info("Compare checksums of files in source vendor folder")
         assert files_checksums == expected_files_checksums
+
+
+@pytest.mark.parametrize(
+    "test_params",
+    [
+        # Test case checks fetching retrodep dependencies, generating environment vars file,
+        # building image with all prepared prerequisites and printing help message for retrodep
+        # app in built image
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/retrodep.git",
+                ref="12a0692be09fc18ce82a71904562d8408fe2296a",
+                packages=({"path": ".", "type": "gomod"},),
+                check_vendor_checksums=False,
+                install_app="retrodep",
+                expected_rc=0,
+                expected_output="All dependencies fetched successfully",
+            ),
+            id="gomod_e2e_test",
+        ),
+    ],
+)
+def test_e2e(
+    test_params: utils.TestParameters,
+    cachi2_image: utils.ContainerImage,
+    tmpdir: Path,
+    test_data_dir: Path,
+    request,
+):
+    """
+    End to end test for package manages.
+
+    :param test_params: Test case arguments
+    :param tmpdir: Temp directory for pytest
+    """
+    test_case = request.node.callspec.id
+
+    source_folder = utils.clone_repository(
+        test_params.repo, test_params.ref, f"{test_case}-source", tmpdir
+    )
+
+    output_folder = utils.fetch_deps_and_check_output(
+        tmpdir, test_case, test_params, source_folder, test_data_dir, cachi2_image
+    )
+
+    log.info("Create cachi2.env file")
+    env_vars_file = os.path.join(tmpdir, "cachi2.env")
+    cmd = ["generate-env", output_folder, "--output", env_vars_file, "--for-output-dir", "/tmp"]
```

Comment:
I think the output-dir is incorrect here, it should be `/tmp/gomod_e2e_test-output`.